pkg/proxy/record.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// FlowDirection is the type to indicate the flow direction
+type FlowDirection string
+
+const (
+	// TypeRequest is a request message
+	TypeRequest = "Request"
+
+	// TypeResponse is a response to a request
+	TypeResponse = "Response"
+)
+
+// FlowVerdict is the verdict taken on request/response
+type FlowVerdict string
+
+const (
+	// VerdictForwared indicates that the request/response was forwarded
+	VerdictForwared = "Forwarded"
+
+	// VerdictDenied indicates that the request/response was denied
+	VerdictDenied = "Denied"
+
+	// VerdictError indicates that there was an error processing the request/response
+	VerdictError = "Error"
+)
+
+// EndpointInfo contains information about the endpoint sending/receiving the
+// request/response
+type EndpointInfo struct {
+	ID       uint64
+	IPv4     string
+	IPv6     string
+	Identity uint64
+	Labels   []string
+}
+
+// NodeInfo holds addressing information of the node the agent runs on
+type NodeInfo struct {
+	IPv4 string
+	IPv6 string
+}
+
+// LogRecord is the structure used to log individual request/response
+// processing events
+type LogRecord struct {
+	// Direction is the direction of the flow
+	Direction FlowDirection
+
+	// StartTime is the timestamp when the flow has started. See Duration
```

Comment:
Please specify the format as RFC3339Nano. The timestamp must be in UTC and include fraction of a second.
For e.g. time.Now().UTC().Format(time.RFC3339Nano) returns 2017-06-08T21:21:04.698805823Z.